pkg/cmd/repo/fork/fork_test.go

```diff
@@ -663,78 +664,111 @@ func TestRepoFork(t *testing.T) {
 			},
 			wantErrOut: "✓ Created fork OWNER/REPO\n✓ Renamed fork to OWNER/NEW_REPO\n",
 		},
+		{
+			name: "retries clone up to four times if necessary",
+			opts: &ForkOptions{
+				Repository: "OWNER/REPO",
+				Clone:      true,
+				BackOff:    &backoff.ZeroBackOff{},
+			},
+			httpStubs: forkPost,
+			execStubs: func(cs *run.CommandStubber) {
+				cs.Register(`git clone https://github.com/someone/REPO\.git`, 128, "")
+				cs.Register(`git clone https://github.com/someone/REPO\.git`, 128, "")
+				cs.Register(`git clone https://github.com/someone/REPO\.git`, 128, "")
+				cs.Register(`git clone https://github.com/someone/REPO\.git`, 0, "")
+				cs.Register(`git -C REPO remote add upstream https://github\.com/OWNER/REPO\.git`, 0, "")
+				cs.Register(`git -C REPO fetch upstream`, 0, "")
+			},
+		},
+		{
+			name: "does not retry clone if error occurs and exit code is not 128",
+			opts: &ForkOptions{
+				Repository: "OWNER/REPO",
+				Clone:      true,
+				BackOff:    &backoff.ZeroBackOff{},
+			},
+			httpStubs: forkPost,
+			execStubs: func(cs *run.CommandStubber) {
+				cs.Register(`git clone https://github.com/someone/REPO\.git`, 128, "")
+				cs.Register(`git clone https://github.com/someone/REPO\.git`, 65, "")
+			},
+			wantErr: true,
+			errMsg:  `failed to clone fork: failed to run git: git -c credential.helper= -c credential.helper=!"[^"]+" auth git-credential clone https://github.com/someone/REPO\.git exited with status 65`,
+		},
 	}
 
 	for _, tt := range tests {
-		ios, _, stdout, stderr := iostreams.Test()
-		ios.SetStdinTTY(tt.tty)
-		ios.SetStdoutTTY(tt.tty)
-		ios.SetStderrTTY(tt.tty)
-		tt.opts.IO = ios
+		t.Run(tt.name, func(t *testing.T) {
```

Comment:
Something that I had not thought about until now is that when using the `-run` flag with `go test` that all this set up happens for each test even though the tests won't be run. If we encapsulate all the code in `t.Run` the setup won't happen unless the test is actually going to be run.